### PR TITLE
Warn and crashloop on daemon registration endpoint deprecation

### PR DESCRIPTION
**Pull request by awh**
Facilitate deprecation of old daemon registration endpoints by having fluxd log that an upgrade is required and crashloop to attract attention from the user.

### CODE REVIEW

File: http/daemon.go [hunk 1]
```diff
@@ -32,7 +33,8 @@ type Daemon struct {
 }
 
 var (
-	connectionDuration = prometheus.NewGaugeFrom(stdprometheus.GaugeOpts{
+	ErrEndpointDeprecated = errors.New("Your fluxd version is deprecated - please upgrade")
```
Comment (awh) [inline http/daemon.go]: Could use `ErrorDeprecated` from `http/errors.go` here, but it is multiline (IMO not great for printing to a daemon log) and mentions fluxctl specifically - thoughts?

Comment (squaremo) [inline http/daemon.go]: This is fine -- I agree with your rationale.

**Comment by squaremo** [comment]
I suspect this has upset a mock implementation somewhere (i.e., the CircleCI failure is not a flake).

File: http/daemon.go (hunk 1 above)
Comment (paulbellamy) [inline http/daemon.go]: instructions on how to upgrade?

**Comment by awh** [comment]
> instructions on how to upgrade?

**Pull request merged**